apps/server/src/modules/fwu-learning-contents/uc/fwu-learning-contents.uc.ts

```diff
@@ -0,0 +1,50 @@
+import { Injectable, NotFoundException, InternalServerErrorException, Inject } from '@nestjs/common';
+import { GetObjectCommand, S3Client } from '@aws-sdk/client-s3';
+import { Readable } from 'stream';
+import { Logger } from '@src/core/logger';
+import { S3Config } from '../interface/config';
+
+@Injectable()
+export class FwuLearningContentsUc {
+	constructor(
+		private logger: Logger,
+		@Inject('FWU_S3_Client') readonly client: S3Client,
+		@Inject('S3_Config') readonly config: S3Config
+	) {
+		this.logger.setContext(FwuLearningContentsUc.name);
+	}
+
+	private async getResponse(path: string) {
+		const request = new GetObjectCommand({
+			Bucket: this.config.bucket,
+			Key: path.toString(),
+		});
+		const response = await this.client.send(request).catch((error: unknown) => {
+			if (error && typeof error === 'object' && 'name' in error && 'stack' in error) {
+				if (error.name === 'NoSuchKey') {
+					throw new NotFoundException({ cause: error.stack });
+				} else {
+					throw new InternalServerErrorException({ name: error.name, cause: error.stack });
+				}
+			}
+			throw new InternalServerErrorException({ name: undefined, cause: error });
+		});
+		return response;
+	}
+
+	async get(path: string): Promise<Uint8Array> {
+		const response = await this.getResponse(path);
```

Comment:
do we have to do any authorisation here? Is access to the concrete files being checked by the S3 client?
even if it is, I think we should at least have a simple permission check in place for a permission that if necessary we can give to all users, but also easily take away if necessary.

Reply:
At this point, the permissions should not be necessary.
The content is available to all students and teachers who can log into the Brandenburg instance.
In the controller, authentication is carried out via jwt and the feature flag ensures that the reuquest is only handled on the Brandenburg instance.
In addition, the fwu-learning-contents microservice only runs on the Brandenburg instance. 
It is not available on the other instances.